presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PruneJoinColumns.java

```diff
@@ -49,38 +47,31 @@ public Pattern getPattern()
     @Override
     public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator, Session session)
     {
-        ProjectNode parent = (ProjectNode) node;
+        return pushDownProjectOff(
```

Comment:
For most node types, it needs to be a lambda, because the lambda captures `idAllocator`, because Java doesn't have a convenient type for a three-argument function object (at least that I can find!).  The lambda does exactly the same thing as the abstract protected member function in ProjectOffPushDownRule.